Source/Scene/Cesium3DTilesetTraversal.js

```diff
@@ -0,0 +1,823 @@
+        // Sort by farthest child first since this is going on a stack
+        if (b.distanceToCamera === 0 && a.distanceToCamera === 0) {
+            return b._centerZDepth - a._centerZDepth;
+        }
+
+        return b.distanceToCamera - a.distanceToCamera;
+    }
+
+    var emptyArray = freezeObject([]);
+
+    function BaseTraversal() {
+        this.tileset = undefined;
+        this.frameState = undefined;
+        this.outOfCore = undefined;
+        this.stack = new ManagedArray();
+        this.leaves = new ManagedArray();
+        this.baseScreenSpaceError = undefined;
+        this.internalDFS = new InternalBaseTraversal();
+    }
+
+    BaseTraversal.prototype.execute = function(tileset, root, baseScreenSpaceError, frameState, outOfCore) {
+        this.tileset = tileset;
+        this.frameState = frameState;
+        this.outOfCore = outOfCore;
+        this.leaves.length = 0;
+        this.baseScreenSpaceError = Math.max(baseScreenSpaceError, this.tileset._maximumScreenSpaceError);
+        this.internalDFS.tileset = this.tileset;
+        this.internalDFS.frameState = this.frameState;
+        this.internalDFS.outOfCore = this.outOfCore;
+        this.internalDFS.baseScreenSpaceError = this.baseScreenSpaceError;
+        DFS(root, this);
+    };
+
+    BaseTraversal.prototype.visitStart = function(tile) {
+        if (tile._lastVisitedFrame !== this.frameState.frameNumber) {
+            visitTile(this.tileset, tile, this.frameState, this.outOfCore);
+        }
+    };
+
+    BaseTraversal.prototype.visitEnd = function(tile) {
+        tile._lastVisitedFrame = this.frameState.frameNumber;
+    };
+
+    BaseTraversal.prototype.getChildren = function(tile) {
+        if (baseUpdateAndCheckChildren(this.tileset, tile, this.baseScreenSpaceError, this.frameState)) {
+            var children = tile.children;
+            var childrenLength = children.length;
+            var allReady = true;
+            var replacementWithContent = tile.refine === Cesium3DTileRefine.REPLACE && tile.hasRenderableContent;
+            for (var i = 0; i < childrenLength; ++i) {
+                var child = children[i];
+                loadTile(child, this.frameState);
+                touch(this.tileset, child, this.outOfCore);
+
+                // content cannot be replaced until all of the nearest descendants with content are all loaded
+                if (replacementWithContent) {
+                    if (!child.hasEmptyContent) {
+                        allReady = allReady && child.contentReady;
+                    } else {
+                        allReady = allReady && this.internalDFS.execute(child);
+                    }
+                }
+            }
+
+            if (allReady) {
+                return children;
+            }
+        }
+        return emptyArray;
+    };
+
+    function baseUpdateAndCheckChildren(tileset, tile, baseScreenSpaceError, frameState) {
+        if (tile.hasTilesetContent) {
```

Comment:
I think you're right, but I think we should then add to the condition below

`tile._screenSpaceError <= baseScreenSpaceError && !tile.hasTilesetContent`

so that the base traversal doesn't stop at the root of external tilesets